app/src/main/java/OptionsManager.java

```diff
@@ -0,0 +1,136 @@
+import com.raylib.Jaylib;
+import com.raylib.Raylib;
+import org.bytedeco.javacpp.IntPointer;
+import org.bytedeco.javacpp.Pointer;
+
+import java.util.concurrent.Callable;
+
+import static com.raylib.Jaylib.*;
+import static com.raylib.Raylib.*;
+
+/**
+ * Die OptionsManager-Klasse verwaltet die Optionen des Spiels.
+ * Singleton Pattern wird benutzt
+ */
+public class OptionsManager {
+
+    // Instanzvariable für das Singleton-Muster
+    private static final OptionsManager instance = new OptionsManager();
+
+    // Anzeige der Optionen
+    boolean show;
+
+    // Optionen für das Spiel
+    float roadWidth;
+    float cameraHeight;
+    float drawDistance;
+    float fogDensity;
+    float fieldOfView;
+
+    // Dropdown-Fenster-Status
+    boolean dropdownWindowOpened;
+
+    // Status der Schaltflächen im Fenster
+    boolean windowButton1Clicked;
+
+    // Rechtecke für GUI-Elemente
+    Jaylib.Rectangle laneTitle = new Jaylib.Rectangle(500, 150, 200, 50);
+    Jaylib.Rectangle laneButtonSize1 = new Jaylib.Rectangle(500, 200, 200, 50);
+    Jaylib.Rectangle laneButtonSize2 = new Jaylib.Rectangle(500, 250, 200, 50);
+    Jaylib.Rectangle laneButtonSize3 = new Jaylib.Rectangle(500, 300, 200, 50);
+    Jaylib.Rectangle laneButtonSize4 = new Jaylib.Rectangle(500, 350, 200, 50);
+
+    Jaylib.Rectangle roadWidthSize = new Jaylib.Rectangle(120, 200, 300, 50);
+    Jaylib.Rectangle cameraHeightSize = new Jaylib.Rectangle(120, 250, 300, 50);
+    Jaylib.Rectangle drawDistanceSize = new Jaylib.Rectangle(120, 300, 300, 50);
+    Jaylib.Rectangle fieldOfViewSize = new Jaylib.Rectangle(120, 350, 300, 50);
+    Jaylib.Rectangle fogDensitySize = new Jaylib.Rectangle(120, 400, 300, 50);
+
+    // Private Konstruktor für das Singleton-Muster
+    private OptionsManager() {
+        // Standardwerte für die Optionen
+        roadWidth = Constants.ROADWIDTH;
+        cameraHeight = Constants.CAMERAHEIGHT;
+        drawDistance = Constants.DRAWDISTANCE;
+        fieldOfView = (float) Constants.FOV;
+        fogDensity = Constants.FOGDENSITY;
+    }
+
+    /**
+     * Gibt die einzige Instanz der OptionsManager-Klasse zurück.
+     *
+     * @return Instanz von OptionsManager
+     */
+    public static OptionsManager getInstance() {
+        return instance;
+    }
+
+    /**
+     * Aktualisiert die Optionen basierend auf Benutzereingaben.
+     */
+    public void update() {
+        // Tastendruck 'S' zeigt/versteckt die Optionen
+        if (IsKeyPressed(KEY_S)) {
+            show = !show;
+        }
```

Comment:
Dieses if würde ich lieber in Game verschieben

Reply:
Findest du es nicht besser, wenn es die Klasse selber entscheidet, welche Taste gedrückt werden muss?

Reply:
In der Game Klasse werden die anderen Keys abgefragt deswegen dachte ich ist übersichtlicher wenn dort alle Keys geprüft werden. So wie es jetzt ist, ist es versteckt in er OptionsManager Klasse.

Reply:
OK, dann ändere ich das auch mal.